### PR TITLE
Issue 27 - Don't make a tile exception break other tiles

### LINKED ISSUE #27: An exception on a tile results on a completely broken page
In this line you raise an exception https://github.com/plone/plone.app.blocks/blob/1c2e7256c04c86aa55beebd7707f4dee92597ca2/plone/app/blocks/utils.py#L103 and because of that if one tile has an error, all of the tiles become broken.

This happens because if the first tile gets an error, all subsequent rendered tiles gets the error. Something like this:

broken

Wouldn't it be the case to return an string error, like we had in the past with portlets?

> There was an error while rendering the portlet

> There was an error while rendering this tile

@hvelarde this relates to collective.cover as well.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #27

### CODE REVIEW

File: plone/app/blocks/utils.py [hunk 1]
```diff
@@ -100,7 +101,7 @@ def resolveResource(url):
 
     response = subrequest(url)
     if response.status == 404:
-        raise NotFound(url)
+        return PloneMessageFactory('There was an error while rendering this tile')
```
Comment (rodfersou) [inline plone/app/blocks/utils.py]: Can't use the ``_`` alias because ``_`` is used above as variable

**Comment by rodfersou** [comment]
I can't figure out how to test this.. can someone help me please?

**Comment by hvelarde** [comment]
@plone/framework-team can somebody review and merge this, please?

**Comment by datakurre** [comment]
Working on it.

**Comment by datakurre** [comment]
@rodfersou @hvelarde I added test and moved the error message into tile merge function (plone.app.blocks.tiles.renderTiles) as I recommended in https://github.com/plone/plone.app.blocks/issues/27 (the originally proposed method was too generic and also used for fetching other resources than tiles).

**Comment by datakurre** [comment]
When tests pass this should be ready for merge.

**Comment by hvelarde** [comment]
thanks, @datakurre! can we have a new release?

**Comment by datakurre** [comment]
@hvelarde Later today! I need to fix dexteritytextindexer suppor to avoid related regression note in changelog :) https://github.com/plone/plone.app.blocks/pull/29

**Comment by datakurre** [comment]
@hvelarde plone.app.blocks 3.1.0 is released; If you need this patch for earlier release, please do a branch with required cherry-picks and I can make a bugfix release.

**Comment by hvelarde** [comment]
@datakurre can we use 3.x branch on collective.cover? what are the incompatibilities?

**Comment by datakurre** [comment]
@hvelarde It changes the included view name for layout behavior (do not affect) and includes indexer for layout field (may affect). Otherwise there should only be additions (all unnecessary for cover).

**Pull request merged**